src/extension/src/ActionHandler.py

```diff
@@ -143,23 +180,97 @@ def copy_config_files(self, src, dst, raise_if_not_copied=False):
         self.logger.log("All required files from the preceding extension version were copied to current one")
 
     def uninstall(self):
-        self.logger.log("Extension uninstalled")
-        return Constants.ExitCode.Okay
+        stdout_file_mirror = None
+        file_logger = None
+        try:
+            log_file_name = datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S") + "_Uninstall"
+            file_logger = self.utility.create_log_file(self.ext_env_handler.log_folder, log_file_name)
+            if file_logger is not None:
+                self.logger.file_logger = file_logger
+                stdout_file_mirror = StdOutFileMirror(file_logger)
+            self.logger.log("Extension uninstalled")
+            return Constants.ExitCode.Okay
+
+        except Exception as error:
+            self.logger.log_error("Error occurred during extension uninstall. [Error={0}]".format(repr(error)))
+            return Constants.ExitCode.HandlerFailed
+
+        finally:
+            if stdout_file_mirror is not None:
+                stdout_file_mirror.stop()
+            if file_logger is not None:
+                file_logger.close()
 
     def enable(self):
-        self.logger.log("Enable triggered on extension")
-        enable_command_handler = EnableCommandHandler(self.logger, self.utility, self.runtime_context_handler, self.ext_env_handler, self.ext_config_settings_handler, self.core_state_handler, self.ext_state_handler, self.ext_output_status_handler, self.process_handler, self.cmd_exec_start_time, self.seq_no)
-        return enable_command_handler.execute_handler_action()
+        stdout_file_mirror = None
+        file_logger = None
+        try:
+            log_file_name = datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S") + "_Enable"
+            file_logger = self.utility.create_log_file(self.ext_env_handler.log_folder, log_file_name)
```

Comment:
Considering that enables can happen more frequently even without any action from CRP, consider if just for enable, using this approach alone will result in other logs getting groomed out when the count increases. As-is is okay for this review, but let's discuss next week if this can be improved on. Develop your thoughts on this and plan on taking a direction here in a separate PR next week.

Reply:
Consider grooming Core logs separately from ext logs. Overflowing ext logs should not groom out core logs.